src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -362,53 +246,35 @@ public boolean logOut(boolean showPrompt) {
       }
     }
 
-    email = "";
-    isLoggedIn = false;
-
+    accountRoster.clear();
     authDataStore.clearStoredOAuthData();
 
-    notifyLoginStatusChange(false);
+    notifyLoginStatusChange();
     uiFacade.notifyStatusIndicator();
     return true;
   }
 
-  public Credential makeCredential() {
-    Credential cred =
-        new GoogleCredential.Builder()
-            .setJsonFactory(jsonFactory)
-            .setTransport(transport)
-            .setClientSecrets(clientId, clientSecret)
-            .build();
-    cred.setAccessToken(accessToken);
-    cred.setRefreshToken(refreshToken);
-    return cred;
-  }
-
-  /**
-   * Performs the firing of listeners and the updates to the UI that are normally performed as part
-   * of a log in or log out, and retrieves any persistently stored credentials upon log in, but does
-   * not actually interact with an OAuth server. This method is provided for use in tests.
-   *
-   * @param login
-   *     {@code true} if a log in is to be simulated, {@code false} if a log out is to be simulated
-   */
-  public void simulateLoginStatusChange(boolean login) {
-    if (login) {
-      retrieveSavedCredentials();
-    }
-    notifyLoginStatusChange(login);
-    uiFacade.notifyStatusIndicator();
+  private Credential makeCredential(
+      String accessToken, String refreshToken, long expiresInSeconds) {
+    Credential credential = new GoogleCredential.Builder()
+        .setJsonFactory(jsonFactory)
+        .setTransport(transport)
+        .setClientSecrets(clientId, clientSecret)
+        .build();
+    credential.setAccessToken(accessToken)
+              .setRefreshToken(refreshToken)
+              .setExpiresInSeconds(expiresInSeconds);
+    return credential;
   }
 
-  private void updateLoginState(GoogleTokenResponse tokenResponse) {
-    refreshToken = tokenResponse.getRefreshToken();
-    accessToken = tokenResponse.getAccessToken();
-    oAuth2Credential = makeCredential();
-    accessTokenExpiryTime = System.currentTimeMillis() / 1000 + tokenResponse.getExpiresInSeconds();
-    email = queryEmail();
-    persistCredentials();
-    uiFacade.notifyStatusIndicator();
-    notifyLoginStatusChange(true);
+  private void updateLoginState(GoogleTokenResponse tokenResponse)
+      throws IOException, EmailAddressNotReturnedException {
+    Credential credential = makeCredential(tokenResponse.getAccessToken(),
+                                           tokenResponse.getRefreshToken(),
+                                           tokenResponse.getExpiresInSeconds());
+    String email = queryEmail(credential);
+    Long expiryTime = System.currentTimeMillis() / 1000 + tokenResponse.getExpiresInSeconds();
```

Comment:
I just realized that we don't need a `Clock` anymore since `Credential` computes and set the time for us. `System.currentTimeMillis()` will be removed from our code.